### ISSUE TITLE
[workflow-archiver-job/prod] StandardError: Lockfile found: archiver may have malfunctioned

### ISSUE THREAD

**Issue by jcoyne** [sul-dlss/workflow-archiver-job#39]
## Backtrace

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/52893/faults/35753441)

**Comment by jcoyne** [comment]
I believe this is because this is kicked off via a cron job every 15 minutes and, the previous job has not yet completed.

**Comment by blalbrit** [comment]
@jcoyne - that is correct. The workflow archiver is run by cron every 15 minutes. An object is not versionable until the workflows have been archived.

When we first set this up, there was much less regular activity in the pipeline - so the check for a lockfile was in place to keep the archiver from spinning up multiple jobs if a very large tranche of objects were already being processed. (oh yes - the archiver took down the entire system several times by starting a new job every 15 minutes regardless of how many other jobs were already running - the lockfile check was intended to keep this from happening).

Because the current state of SDR can not handle multiple versions in the pipeline at the same time, this process is required to complete before new object versions can be opened by a user.

We now have a lot more content in the system because of higher use by multiple stakeholders. Until SDR improvements happen in the new year, we can:
* continue to use the lockfile check to avoid multiple archiver threads spinning up and recharacterize this error
* change the frequency that the crons are run (30 mins might produce fewer errors, but would come with a trade-off that objects would always spend longer in a non-versionable state, regardless of amount of content in the system)
* introduce a new check for malfunction somehow (not sure what that would look like)

I'm not entirely sure which path would be preferable at this point - thoughts @jcoyne ?

**Comment by jcoyne** [comment]
@blalbrit is there some reason we don't put a message in resque when a workflow can be archived and just act on that ASAP?

**Comment by jcoyne** [comment]
Also, what do we do with the data once its in the archive table?

**Comment by blalbrit** [comment]
the data in the archive table is retained for provenance information (and to feed the workflows datastream I think - but we should confirm that).

the only reason I can think of re: active messaging for archiving is that it probably wasn't in the team's skillset when this was implemented almost a decade ago. it totally makes sense to think about revisiting that approach.  I think that the whole workflow/archiving/Oracle situation is on the table for an early 2019 work cycle. Worth thinking about changing this then?

**Comment by jcoyne** [comment]
Yes, that's why I'm trying to understand what is going on, so that we can think about changing it.  Vivian has asked us to familiarize ourselves with some of the potential work for the upcoming work cycle.

**Comment by jcoyne** [comment]
Do you know how/what moves the archived workflow information to the workflows datastream? Is there a way a user can access the provenance information?

**Comment by blalbrit** [comment]
^^ I think this: https://github.com/sul-dlss/dor-services/blob/39eb6012a7b93fe2b0059d90410e0fc7f8c33813/lib/dor/datastreams/workflow_ds.rb

**Comment by blalbrit** [comment]
Users can view the workflow history, but shouldn't have a way to alter it, through Argo.
